src/java/org/jivesoftware/openfire/plugin/service/LogAPI.java

```diff
@@ -62,7 +62,7 @@ public Response getLoggedServiceNames()
 
         final List<MultiUserChatService> multiUserChatServices = XMPPServer.getInstance().getMultiUserChatManager().getMultiUserChatServices();
         final List<String> serviceNames = multiUserChatServices.stream()
-            .filter(s -> s.getChatRooms().stream().anyMatch(r -> r.isLogEnabled() && r.isPublicRoom()))
+            .filter(s -> s.getActiveChatRooms().stream().anyMatch(r -> r.isLogEnabled() && r.isPublicRoom()))
```

Comment:
I don't think that using only the _active_ chat rooms is quite correct here. The functionality that is implemented here is exposing logs. It is perfectly valid for this to be used for chat rooms that have long ago stopped by active. Maybe switch to `getActiveAndInactiveRooms()` (or find a way to first evaluate the active rooms, and if none match also evaluate the inactive ones, for a bit of optimization - if that is at all feasible).

Reply:
getActiveChatRooms calls the previous deprecated method and so retains the previous behaviour.

That being said, just because it's what the old code did, that might not be what the old code was intended to do.

There isn't a getInactiveChatRooms, but there is  a getAllRoomNames that we can use to then get the props of all rooms. Oddly, using that once would likely mean that all rooms are now active rooms. 

Given some environments with many many rooms, if we want to change this, do we need to give good thought about memory efficiency in this operation? A plain map of the stream of names to MUCRooms could be slow and use a lot of memory.

Reply:
I believe that `getActiveAndInactiveRooms()` (at least [on the main branch](https://github.com/igniterealtime/Openfire/blob/main/xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java#L2015-L2034)) does not use the old deprecated methods (if only because those got removed from the code).

You are right that this is going to be resource intensive, on large setups. The old code was too, I think, so I don't think we're making things worse. Optimizing things (if at all possible) could be a separate effort.

Reply:
I was looking at the Service, not the Impl, but the same argument applies. The LocalMUCRoomManager used by the Impl will have the same issue due to the CleanupTask, that it only returns the stuff that's in the cache, rather than everything.

We could either:
* Keep it as active rooms, validate the behaviour is as we suspect, create an issue and fix (with optimisation) later
* switch this to all names, fix the behaviour now, assume a performance impact and fix that via optimisation later

I'm in favour of the first, given the scope of the PR (perhaps with a code comment to say that it's probably wrong)

Reply:
Ok, let's go with that first option.